guides/keras_cv/object_detection_keras_cv.py

```diff
@@ -0,0 +1,868 @@
+
+"""
+Looks like everything is structured as expected.
+Now we can move on to constructing our
+data augmentation pipeline.
+
+## Data augmentation
+
+One of the most challenging tasks when constructing object detection
+pipelines is data augmentation.  Image augmentation techniques must be aware of the underlying
+bounding boxes, and must update them accordingly.
+
+Luckily, KerasCV natively supports bounding box augmentation with its extensive
+library
+of [data augmentation layers](https://keras.io/api/keras_cv/layers/preprocessing/).
+The code below loads the Pascal VOC dataset, and performs on-the-fly bounding box
+friendly data augmentation inside of a `tf.data` pipeline.
+"""
+augmenter = keras.Sequential(
+    layers=[
+        keras_cv.layers.RandomFlip(mode="horizontal", bounding_box_format="xywh"),
+        keras_cv.layers.JitteredResize(
+            target_size=(640, 640), scale_factor=(0.75, 1.3), bounding_box_format="xywh"
+        ),
+    ]
+)
+
+train_ds = train_ds.map(augmenter, num_parallel_calls=tf.data.AUTOTUNE)
+visualization.visualize_dataset(train_ds, bounding_box_format="xywh", value_range=(0, 255), rows=2, cols=2)
+
+"""
+Great!  We now have a bounding box friendly data augmentation pipeline.
+Let's format our evaluation dataset to match.  Instead of using
+`JitteredResize`, let's use the deterministic `keras_cv.layers.Resizing()`
+layer.
+"""
+
+inference_resizing = keras_cv.layers.Resizing(
+    640, 640, bounding_box_format="xywh", pad_to_aspect_ratio=True
+)
+eval_ds = eval_ds.map(inference_resizing, num_parallel_calls=tf.data.AUTOTUNE)
+
+"""
+Due to the fact that the resize operation differs between the train dataset,
+which uses `JitteredResize()` to resize images, and the inference dataset which
+uses `layers.Resizing(pad_to_aspect_ratio=True)`, it is good practice to
+visualize both datasets:
+"""
+
+visualization.visualize_dataset(eval_ds, bounding_box_format="xywh", value_range=(0, 255), rows=2, cols=2)
+
+"""
+Finally, let's unpackage our inputs from the preprocessing dictionary, and
+prepare to feed the inputs into our model.  In order to be TPU compatible,
+bounding box Tensors need to be `Dense` instead of `Ragged`.  If training on
+GPU, you can omit the `bounding_box.to_dense()` call.  If ommitted,
+the KerasCV RetinaNet
+label encoder will automatically correctly encode Ragged training targets.
+"""
+
+def dict_to_tuple(inputs):
+    return inputs["images"], bounding_box.to_dense(
+        inputs["bounding_boxes"], max_boxes=32
+    )
+
+train_ds = train_ds.map(dict_to_tuple, num_parallel_calls=tf.data.AUTOTUNE)
+eval_ds = eval_ds.map(dict_to_tuple, num_parallel_calls=tf.data.AUTOTUNE)
+
+train_ds = train_ds.prefetch(tf.data.AUTOTUNE)
+eval_ds = eval_ds.prefetch(tf.data.AUTOTUNE)
+
+"""
+Our data pipeline is now complete!
+We can now move on to model creation and training.
+
+### Model creation
```

Comment:
Should be a lvl-2 section (or create an encompassing lvl-2 section for "model architecture", "optimizer", "loss functions", "metrics")

Reply:
I think the untrained model creation should occur after you discuss loss and metrics on the pretrained model